src/purchases.ts

```diff
@@ -945,4 +1023,20 @@ export default class Purchases {
         }
     }
 
+    private static async throwIfAndroidPlatform() {
+        if (Platform.OS === "android") {
+            throw new UnsupportedPlatformError()
```

Comment:
I mentioned in the flutter PR but just in case, we used to just not do anything when the platform was unsupported, but we are throwing an error here. The alternative would be to make the return type of these new methods nullable. But since users should do some error handling in any case, I think this is ok.

Reply:
I like this